src/RECONR/ReferenceGrid/src/call.hpp

```diff
@@ -1,65 +1,172 @@
-std::vector< double > operator()( const RP::SpecialCase& sc ) const {
-  return { sc.EL(), sc.EH() };
+std::array<double, 3>
+operator()( const resolved::BreitWigner::Lvalue::Resonance& resonance ) const {
+  const double energy = resonance.ER();
+  const double halfWidth = 0.5 * resonance.GT();
+  return {{ energy - halfWidth, energy, energy + halfWidth }};
 }
 
-std::array<double, 3> operator()( 
-  const RP::resolved::BreitWigner::Lvalue::Resonance& resonance ) const {
+std::array<double, 3>
+operator()( const resolved::ReichMoore::Lvalue::Resonance& resonance ) const {
+  const double energy = resonance.ER();
+  const double halfWidth = 0.5 * ( resonance.GN()
+                                   + resonance.GG()
+                                   + resonance.GFA()
+                                   + resonance.GFB() );
+  return {{ energy - halfWidth, energy, energy + halfWidth }};
+}
 
-  const double E = resonance.ER();
-  const double hw = resonance.GT()*0.5;
-  return {{E-hw, E, E+hw}};
+template< typename ResolvedLvalue,
+          if_< hasResonances< ResolvedLvalue > > = true >
```

Comment:
Is this `if_` used to help the compiler know what call operator to use?

Reply:
You can see the definition [here](src/RECONR/ReferenceGrid/meta.hpp)